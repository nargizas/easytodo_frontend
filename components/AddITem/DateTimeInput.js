import { Button, View, Text, StyleSheet, Pressable } from 'react-native';
import { getFormattedDate, getFormattedDateAndTime, getFormattedTime } from "../../util/date";
import { useState } from 'react';
import { GlobalStyles } from "../../constants/styles";
import DateTimePicker from '@react-native-community/datetimepicker'
function DateTimeInput({ date, show, showDatepicker, showTimepicker, dateInputConfig }) {

    return (
        <View style={styles.inputContainer}>

            <Text style={styles.label}>Deadline</Text>
            <View style={{ flexDirection: 'row' }}>
                <Pressable onPress={showDatepicker} style={({ pressed }) => pressed ? [styles.buttonContainer, styles.pressed] : styles.buttonContainer}>
                    <View style={styles.button}>
                        <Text style={styles.buttonText}>Pick date</Text>
                    </View>
                </Pressable>

                <Pressable onPress={showTimepicker} style={({ pressed }) => pressed ? [styles.buttonContainer, styles.pressed] : styles.buttonContainer}>
                    <View style={styles.button}>
                        <Text style={styles.buttonText}>Pick time</Text>
                    </View>
                </Pressable>

            </View>
            <View style={styles.text}>
                <Text style={[styles.inputContainer, styles.buttonText, { color: GlobalStyles.colors.grey }]}>Selected: {getFormattedDateAndTime(date)}</Text>

            </View>
            {
                show && (
                    <DateTimePicker {...dateInputConfig} />
                )
            }
        </View >
    );
}

export default DateTimeInput;

const styles = StyleSheet.create({
    label: {
        fontSize: 18,
        color: GlobalStyles.colors.grey,
        marginBottom: 4,
    },
    inputContainer: {
        marginHorizontal: 16,
        marginVertical: 8,
    },
    buttonContainer: {
        flex: 1
    },
    button: {
        backgroundColor: GlobalStyles.colors.grey,
        paddingVertical: 8,
        justifyContent: 'center',
        alignContent: 'center',
        alignItems: 'center',
        marginHorizontal: 16,
        borderRadius: 16,
    },
    buttonText: {
        fontSize: 16
    },
    pressed: {
        opacity: 0.75
    },
    text: {
        alignItems: 'center'
    }

});

