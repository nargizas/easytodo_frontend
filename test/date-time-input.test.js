import React from 'react';
import renderer from 'react-test-renderer';

import DateTimeInput from '../components/AddITem/DateTimeInput';

it('renders correctly', () => {
  const tree = renderer.create(<DateTimeInput date={new Date(167124561000)} />).toJSON();
  expect(tree).toMatchSnapshot();
});
