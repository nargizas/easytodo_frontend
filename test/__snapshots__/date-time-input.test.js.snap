// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "marginHorizontal": 16,
      "marginVertical": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "#f5f5f5",
        "fontSize": 18,
        "marginBottom": 4,
      }
    }
  >
    Deadline
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        style={
          {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#f5f5f5",
            "borderRadius": 16,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        >
          Pick date
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        style={
          {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#f5f5f5",
            "borderRadius": 16,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        >
          Pick time
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        [
          {
            "marginHorizontal": 16,
            "marginVertical": 8,
          },
          {
            "fontSize": 16,
          },
          {
            "color": "#f5f5f5",
          },
        ]
      }
    >
      Selected:
       
      1975-4-19 16:29
    </Text>
  </View>
</View>
`;
